mersenne-31/src/extension.rs

```diff
@@ -63,28 +79,41 @@ impl BinomiallyExtendable<3> for Mersenne31Complex<Mersenne31> {
     }
 }
 
+impl HasTwoAdicBionmialExtension<3> for Mersenne31Complex<Mersenne31> {
+    const EXT_TWO_ADICITY: usize = 32;
+
+    fn ext_two_adic_generator(bits: usize) -> [Self; 3] {
+        field_to_array::<Self, 3>(Self::two_adic_generator(bits))
+    }
+}
+
 #[cfg(test)]
 mod test_cubic_extension {
 
-    use p3_field_testing::test_field;
+    use p3_field::extension::binomial_extension::BinomialExtensionField;
+    use p3_field_testing::{test_field, test_two_adic_extension_field};
+
+    use crate::{Mersenne31, Mersenne31Complex};
+    type F = Mersenne31Complex<Mersenne31>;
+    type EF = BinomialExtensionField<F, 3>;
+
+    test_field!(super::EF);
 
-    test_field!(
-        p3_field::extension::binomial_extension::BinomialExtensionField<
-            crate::Mersenne31Complex<crate::Mersenne31>,
-            3,
-        >
-    );
+    test_two_adic_extension_field!(super::F, super::EF);
 }
 
 #[cfg(test)]
 mod test_quadratic_extension {
 
-    use p3_field_testing::test_field;
+    use p3_field::extension::binomial_extension::BinomialExtensionField;
+    use p3_field_testing::{test_field, test_two_adic_extension_field};
+
+    use crate::{Mersenne31, Mersenne31Complex};
+
+    type F = Mersenne31Complex<Mersenne31>;
+    type EF = BinomialExtensionField<F, 2>;
+
+    test_field!(super::EF);
 
-    test_field!(
-        p3_field::extension::binomial_extension::BinomialExtensionField<
-            crate::Mersenne31Complex<crate::Mersenne31>,
-            2,
-        >
-    );
+    test_two_adic_extension_field!(super::F, super::EF);
```

Comment:
Should these crates also invoke `test_two_adic_field!`?

Reply:
Yes, add within the macro.